app/javascript/controllers/patron_request_controller.js

```diff
@@ -34,8 +34,30 @@ export default class extends Controller {
     this.accordionTargets.filter(e => e.dataset.patronrequestForrequesttype).forEach(el => {
       if (el.dataset.patronrequestForrequesttype == requestType) {
         el.classList.remove('d-none');
+        this.enableRequiredInputs(el);
       } else {
         el.classList.add('d-none');
+        this.disableRequiredInputs(el);
+      }
+    });
+  }
+
+  // Temporarily disable required inputs that are children of the given element
+  // This is used to prevent validation errors when hidden/unused inputs are required
+  // The initial required state of the inputs is preserved via a data attribute
+  disableRequiredInputs(element) {
+    element.querySelectorAll('input, select, textarea').forEach(input => {
```

Comment:
This can be replaced with querySelectorAll('[required]') and you won't need the if statement

Reply:
that's smart! i was able to do it for the other one too.